shell/platform/fuchsia/flutter/accessibility_bridge.cc

```diff
@@ -927,4 +885,67 @@ void AccessibilityBridge::FillInspectTree(int32_t flutter_node_id,
 }
 #endif  // !FLUTTER_RELEASE
 
+AccessibilityBridge::FuchsiaAtomicUpdate::FuchsiaAtomicUpdate(
+    std::queue<FuchsiaAtomicUpdate>* atomic_updates)
+    : atomic_updates_(atomic_updates) {}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::AddNodeUpdate(
+    fuchsia::accessibility::semantics::Node node,
+    size_t size) {
+  if (size > kMaxMessageSize) {
+    // TODO(MI4-2531, FIDL-718): Remove this
+    // This is defensive. If, despite our best efforts, we ended up with a node
+    // that is larger than the max fidl size, we send no updates.
+    PrintNodeSizeError(node.node_id());
+    return;
+  }
+  updates_.emplace_back(std::move(node), size);
+}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::AddNodeDeletion(uint32_t id) {
+  deletions_.push_back(id);
+}
+
+void AccessibilityBridge::FuchsiaAtomicUpdate::Apply(
+    fuchsia::accessibility::semantics::SemanticTreePtr* tree) {
```

Comment:
This is probably me being bad at FIDL: why do we need to take a pointer to a pointer here instead of just a single pointer?

Reply:
It is an ownership issue: we want to have the bridge owning the original pointer (FidlVarPtr), but we want the atomic update to be able to call it (to encapsulate logic).